contracts/core/mpl-rewards/v1/test/FarmingWithPoolFdt.t.sol

```diff
@@ -186,108 +176,30 @@ contract PoolCustodialTest is TestUtil {
-
-        assertEq(pool1.balanceOf(address(fez)), depositAmt);
-        assertEq(pool1.balanceOf(address(fox)),          0);
-
-        assertTrue(!fez.try_transfer(address(pool1), address(fox), transferableAmt + 1));  // Fez cannot transfer more than balance - totalCustodyAllowance
-        assertTrue( fez.try_transfer(address(pool1), address(fox),     transferableAmt));  // Fez can transfer transferableAmt
-
-        assertEq(pool1.balanceOf(address(fez)), depositAmt - transferableAmt);
-        assertEq(pool1.balanceOf(address(fox)), transferableAmt);
+        custody_and_transfer(depositAmt, custodyAmt1, custodyAmt2, true, IStakeToken(address(pool1)));
     }
 
     function test_custody_and_withdraw(uint256 depositAmt, uint256 custodyAmt) public {
-        Custodian custodian = new Custodian();
-
-        depositAmt = constrictToRange(depositAmt, 1, 1E9,        true);  // $1 - $1b
-        custodyAmt = constrictToRange(custodyAmt, 1, depositAmt, true);  // $1 - deposit
-
-        mintFundsAndDepositIntoPool(fez, pool1, depositAmt * USD, depositAmt * USD);
-
-        pat.setLockupPeriod(address(pool1), 0);
-
-        assertEq(pool1.custodyAllowance(address(fez), address(custodian)), 0);
-        assertEq(pool1.totalCustodyAllowance(address(fez)),                0);
-
-        fez.increaseCustodyAllowance(address(pool1), address(custodian), custodyAmt * WAD);
-
-        assertEq(pool1.custodyAllowance(address(fez), address(custodian)), custodyAmt * WAD);
-        assertEq(pool1.totalCustodyAllowance(address(fez)),                custodyAmt * WAD);
-
-        uint256 withdrawableAmt = (depositAmt - custodyAmt) * USD;
-
-        assertEq(pool1.balanceOf(address(fez)), depositAmt * WAD);
-
-        make_withdrawable(fez, pool1);
-
-        assertTrue(!fez.try_withdraw(address(pool1), withdrawableAmt + 1));
-        assertTrue( fez.try_withdraw(address(pool1),     withdrawableAmt));
-
-        assertEq(pool1.balanceOf(address(fez)), custodyAmt * WAD);
-        assertEq(usdc.balanceOf(address(fez)), withdrawableAmt);
+        custody_and_pullback(depositAmt, custodyAmt, true, IStakeToken(address(pool1)));
     }
 
     function test_transferByCustodian(uint256 depositAmt, uint256 custodyAmt) public {
-        Custodian custodian = new Custodian();  // Custodial contract for PoolFDTs - will start out as liquidity mining but could be broader DeFi eventually
-
-        depositAmt  = constrictToRange(depositAmt, 1, 1E9,        true);  // $1 - $1b
-        custodyAmt  = constrictToRange(custodyAmt, 1, depositAmt, true);  // $1 - deposit
-
-        mintFundsAndDepositIntoPool(fay, pool1, depositAmt * USD, depositAmt * USD);
-
-        depositAmt  *= WAD;
-        custodyAmt  *= WAD;
+        fdt_transferByCustodian(depositAmt, custodyAmt, true, IStakeToken(address(pool1)));
+    }
 
-        fay.increaseCustodyAllowance(address(pool1), address(custodian), custodyAmt);
+    function test_stake() public {
```

Comment:
Same thing here, we should convert these to fuzz

Reply:
Again do we want it?